search-service/src/main/kotlin/com/egm/stellio/search/service/IAMListener.kt

```diff
@@ -31,31 +31,53 @@ class IAMListener(
+        when (authorizationEvent) {
             is EntityCreateEvent -> createSubjectReferential(authorizationEvent)
             is EntityDeleteEvent -> deleteSubjectReferential(authorizationEvent)
             is AttributeAppendEvent -> updateSubjectProfile(authorizationEvent)
-            is AttributeReplaceEvent -> logger.debug("Not interested in attribute replace events for IAM events")
+            is AttributeReplaceEvent -> "Not interested in attribute replace events for IAM events".right()
             is AttributeDeleteEvent -> removeSubjectFromGroup(authorizationEvent)
-            else -> logger.info("Authorization event ${authorizationEvent.operationType} not handled.")
-        }
+            else -> "Authorization event ${authorizationEvent.operationType} not handled.".right()
+        }.fold({
+            logger.error(
+                "Error while handling event ${authorizationEvent.operationType}" +
+                    " for entity ${authorizationEvent.entityId}: $it"
+            )
+        }, {
+            logger.debug(
+                "Successfully handled event ${authorizationEvent.operationType}" +
+                    " for entity ${authorizationEvent.entityId}"
+            )
+        })
     }
 
     @KafkaListener(topics = ["cim.iam.rights"], groupId = "search-iam-rights")
-    fun processIamRights(content: String) {
+    suspend fun processIamRights(content: String) {
         logger.debug("Received IAM rights event: $content")
-        when (val authorizationEvent = JsonUtils.deserializeAs<EntityEvent>(content)) {
+        val authorizationEvent = JsonUtils.deserializeAs<EntityEvent>(content)
+        when (authorizationEvent) {
             is AttributeAppendEvent -> handleAttributeAppendOnRights(authorizationEvent)
             is AttributeReplaceEvent -> handleAttributeReplaceOnRights(authorizationEvent)
             is AttributeDeleteEvent -> handleAttributeDeleteOnRights(authorizationEvent)
-            else -> logger.info("Authorization event ${authorizationEvent.operationType} not handled.")
-        }
+            else -> "Authorization event ${authorizationEvent.operationType} not handled.".right()
+        }.fold({
+            logger.error(
+                "Error while handling event ${authorizationEvent.operationType}" +
+                    " for entity ${authorizationEvent.entityId}: $it"
+            )
+        }, {
+            logger.debug(
+                "Successfully handled event ${authorizationEvent.operationType}" +
+                    " for entity ${authorizationEvent.entityId}"
+            )
+        })
     }
 
     @KafkaListener(topics = ["cim.iam.replay"], groupId = "search-iam-replay")
-    fun processIamReplay(content: String) {
+    suspend fun processIamReplay(content: String) {
         logger.debug("Received IAM replay event: $content")
         when (val authorizationEvent = JsonUtils.deserializeAs<EntityEvent>(content)) {
```

Comment:
Why we don't use the same configuration than above 
`val authorizationEvent = JsonUtils.deserializeAs<EntityEvent>(content)
 when (authorizationEvent) ...`

And do not add the fold with  logger.error and logger.debug ?

Reply:
because replay is one shot process and I did not want to take care of it...